### PR TITLE
Fix the managed providers retrieval

### PR DESCRIPTION
**Pull request by mfalesni**
So, in October, there was a split in database, splitting some information about providers to a separate table. This needed to be taken care of.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit b230e23a23e2f48bd5e24754f752ef04832de2e0:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@psav Peek request, please :)

**Comment by psav** [comment]
Does ```ext_management_system``` usage need further fixing else where? What exactly happened and moved in the split?

**Comment by mfalesni** [comment]
@psav Yes, the endpoints I think. The IP address column is no longer present in the table.

**Comment by mfalesni** [comment]
Also fixed one caching nastiness :)

File: utils/appliance.py [hunk 1]
```diff
@@ -529,16 +530,25 @@ def managed_providers(self):
...
-            elif ems.hostname is not None:
-                if re.match(r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$", ems.hostname) is not None:
-                    ip_addresses.add(ems.hostname)
+        if self.is_downstream:
+            if self.version >= '5.6':  # It did not make it to 5.5 yet
+                query = self._query_post_endpoints()
+            else:
+                query = self._query_pre_endpoints()
+        else:
+            if self.build_datetime >= datetime(year=2015, month=10, day=5):
```
Comment (psav) [inline utils/appliance.py]: Does this catch upstream too? or is that caught by ```version > 5.6```?

**Comment by dajoRH** [comment]
Lint report for commit 5232b39be9e9205360040a45e4224490e6c9c6f8:

Everything seems all right :smile: .
*CFME QE Bot*

File: utils/appliance.py [hunk 2]
```diff
@@ -529,16 +530,25 @@ def managed_providers(self):
...
         """
-        ems_table = self.db["ext_management_systems"]
         ip_addresses = set([])
-        for ems in self.db.session.query(ems_table):
-            if ems.ipaddress is not None:
-                ip_addresses.add(ems.ipaddress)
-            elif ems.hostname is not None:
-                if re.match(r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$", ems.hostname) is not None:
-                    ip_addresses.add(ems.hostname)
+        if self.is_downstream:
```
Comment (psav) [inline utils/appliance.py]:
```python
        query = self._query_pre_endpoints()
        if self.is_downstream and self.version >= '5.6':  # It did not make it to 5.5 yet
            query = self._query_post_endpoints()
        elif self.build_datetime >= datetime(year=2015, month=10, day=5):
            query = self._query_post_endpoints()
```

**Comment by dajoRH** [comment]
Lint report for commit 3df1a0458066e014e59646536a87b60be768098a:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**